### PR TITLE
[YANG] Add yang model to `SUBNET_DECAP` table

### PR DESCRIPTION
**Pull request by lolyu**
#### Why I did it
Add Yang model to config DB `SUBNET_DECAP` table.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>

##### Work item tracking
- Microsoft ADO **(number only)**: 30473162

#### How I did it
Add the yang model file and UT.

#### How to verify it
Build sonic-yang-model and sonic-yang-mgmt.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-subnet-decap.yang [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+                leaf status {
+					type string {
+						pattern "enable|disable";
+					}
+                    default "disable";
+                    description "Subnet Decap status.";
+                }
+
+                leaf src_ip {
```
Comment (bingwang-ms) [inline src/sonic-yang-models/yang-models/sonic-subnet-decap.yang:41]: Does this also support IPv4 without prefix len? Like `192.168.0.1`?

File: src/sonic-yang-models/yang-models/sonic-subnet-decap.yang [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                    type inet:ipv4-prefix;
+                    description "Subnet decap term source IPv4 prefix.";
+                }
+
+                leaf src_ip_v6 {
```
Comment (bingwang-ms) [inline src/sonic-yang-models/yang-models/sonic-subnet-decap.yang:46]: Same above. And this field must be present.

File: src/sonic-yang-models/tests/files/sample_config_db.json [hunk 3]
```diff
@@ -2762,6 +2762,13 @@
 		"log_verbosity": "notice"
 	    }
 	},
+    "SUBNET_DECAP": {
```
Comment (ganglyu) [inline src/sonic-yang-models/tests/files/sample_config_db.json:2765]: This line is not aligned properly

File: src/sonic-yang-models/yang-models/sonic-subnet-decap.yang [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+                key "name";
+
+                leaf name {
+                    type string;
+                    description "Subnet Decap config name.";
+                }
+
+                leaf status {
+					type string {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-subnet-decap.yang:34]: This line is not aligned, and you can use existing type:
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-templates/sonic-types.yang.j2#L181

**Comment by ganglyu** [comment]
This is a new table, so we need to update https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/doc/Configuration.md

**Review by ganglyu — APPROVED** [review]

File: src/sonic-yang-models/yang-models/sonic-subnet-decap.yang [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+    namespace  "http://github.com/sonic-net/sonic-subnet-decap";
+
+    prefix subnet-decap;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+	import sonic-types {
```
Comment (ganglyu) [inline src/sonic-yang-models/yang-models/sonic-subnet-decap.yang:13]: This line is not aligned properly

File: src/sonic-yang-models/yang-models/sonic-subnet-decap.yang (hunk 1 above)
Comment (lolyu) [reply src/sonic-yang-models/yang-models/sonic-subnet-decap.yang:43]: yes, `192.168.0.1` is regarded as `192.168.0.1/32`

File: src/sonic-yang-models/yang-models/sonic-subnet-decap.yang (hunk 2 above)
Comment (lolyu) [reply src/sonic-yang-models/yang-models/sonic-subnet-decap.yang:49]: Updated, mark this as mandatory.

**Review by ganglyu — APPROVED** [review]

**Review by bingwang-ms — DISMISSED** [review]

**Pull request merged**